src/main/java/org/isf/menu/manager/UserBrowsingManager.java

```diff
@@ -334,37 +313,42 @@ public UserGroup newUserGroup(UserGroup userGroup, List<Permission> permissions)
 		String code = userGroup.getCode();
 		if (ioOperations.isGroupNamePresent(code)) {
 			throw new OHDataIntegrityViolationException(
-					new OHExceptionMessage(MessageBundle.formatMessage("angal.groupsbrowser.thegroupalreadyexists.fmt.msg", code)));
+				new OHExceptionMessage(MessageBundle.formatMessage("angal.groupsbrowser.thegroupalreadyexists.fmt.msg", code)));
 		}
 		return ioOperations.newUserGroup(userGroup, permissions);
 	}
 
 	/**
 	 * Updates an existing {@link UserGroup} in the DB.
-	 *
 	 * @param userGroup - the {@link UserGroup} to update
-	 * @return {@code true} if the group has been updated, {@code false} otherwise.
+	 * @return the {@link UserGroup} that has been updated
+	 * @throws OHServiceException If failed to update group
 	 */
-	public boolean updateUserGroup(UserGroup userGroup) throws OHServiceException {
+	public UserGroup updateUserGroup(UserGroup userGroup) throws OHServiceException {
+		if (userGroup.isDeleted()) {
+			List<User> users = getUser(userGroup.getCode());
+			if (users != null && users.stream().anyMatch(User::isDeleted)) {
+				throw new OHDataValidationException(
+					new OHExceptionMessage(MessageBundle.getMessage("angal.groupsbrowser.thisgrouphasusersandcannotbedeleted.msg")));
+			}
+		}
```

Comment:
this check is already done in `deleteGroup(UserGroup aGroup)`, we don't need it here